explainer.md

```diff
@@ -18,96 +18,107 @@ permanent "block" policies, or
+    contextual flow that helps users understand what's being asked for and why,
+    enabling quick and confident responses. Often, however, permission requests
+    are correlated poorly with user expectations, up to and including prompts
+    that can come out of nowhere (see example 1). This places a burden on user
+    agents' presentation of the request. The user agent has no semantic
+    understanding of events taking place in the content area prior to the
+    permission request. User agents could make better decisions and provide
+    better prompts if they could make well-founded assumptions about the nature
+    of the user's interaction in the content area, and the user's intent.
+
+    ![](images/image1.png) \
+    *Example 1. A notification permission prompt on a news site (contents
+    blurred), shown after the user has clicked on the empty area next to the
+    article content. The user finds this prompt interruptive as they had no
+    interest in subscribing to notifications, and they will likely struggle to
+    understand why the prompt was shown to begin with.*
+
+1.  **Location**: In the ideal case above, users will interact with something on
+    a site that triggers a prompt. In less ideal cases, the user might not have
+    interacted with anything at all, or they may have interacted with an element
+    that was unrelated to the request. Given this uncertainty, user agents rely
+    on common placement of the permission prompt, usually in the top-left of the
+    page. Even in the best case, this has the unfortunate effect of shifting the
+    point to which users need to pay attention from the thing they clicked on to
+    some distant part of the user agent's UI (see example 2). User agents could
+    make better decisions and provide better prompts if they could make
+    well-founded assumptions about the nature of the user's interaction in the
+    content area, and the user's current area of focus. In effect, we think
+    there is a benefit to semantic markup for permissions.
+
+    ![](images/image2.png) \
+    *Example 2. An example where the permission prompt is far away from the
+    user's current area of focus. The permission prompt was triggered because
+    the user has just clicked on the crosshair icon in the bottom right, but the
+    prompt is easy to miss since it's on the opposite side of the page.*
+
+1.  **Regret**: Given the challenges of permission annoyance and abuse, it is
+    reasonable for user agents to suppress a site's future requests for the same
+    capability when the first request is blocked. That said, our research shows
+    that users can and do change their minds for good reasons. When they change
+    their mind, the site can no longer offer an interface in web content and the
+    user must search for the appropriate user agent surface. Our research shows
+    that users often fail when trying to do so (see example 3). In these cases,
```

Comment:
this is true... but again, this is a browser UI flaw that could be addressed at the browser level. 

Chrome itself seems to have attempted to address a bit of this, but bringing site settings directly into the address bar:  

![Screenshot 2023-12-20 at 11 50 31](https://github.com/WICG/PEPC/assets/870154/662ee4c9-f4ef-4dfb-8ef1-9232eccefa11)

There is not reason, why the permission toggles for a site couldn't be shown in that drop down. I guess what I'm trying to say is that we might not have exhausted all UX affordances yet at the browser level. 

I'm worried that we are shifting the problem from browser developers to could end up putting the permissions button in weird and awkward places too... so users would also need to hunt around a site for them and it could be just as bad, if not worse than what browsers do.

Reply:
Extracted into https://github.com/WICG/PEPC/issues/8